### PR TITLE
monitor mobile network state

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

File: shared/actions/platform-specific/index.native.tsx [hunk 1]
```diff
@@ -241,20 +241,29 @@ function* persistRoute(state, action: ConfigGen.PersistRoutePayload) {
...
   const channel = Saga.eventChannel(emitter => {
-    NetInfo.addEventListener('connectionChange', ({type}) => emitter(type === 'none' ? 'offline' : 'online'))
+    NetInfo.addEventListener('connectionChange', ({type}) => emitter(type))
     return () => {}
   }, Saga.buffers.sliding(1))
 
   while (true) {
     const status = yield Saga.take(channel)
-    yield Saga.put(ConfigGen.createOsNetworkStatusChanged({online: status === 'online'}))
+    yield Saga.put(ConfigGen.createOsNetworkStatusChanged({online: status !== 'none'}))
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.native.tsx]: we should just add a parameter here for the type

File: shared/actions/platform-specific/index.native.tsx [hunk 2]
```diff
@@ -241,20 +241,29 @@ function* persistRoute(state, action: ConfigGen.PersistRoutePayload) {
   )
 }
 
+const updateMobileNetState = (type: string) => {
```
Comment (chrisnojima) [inline shared/actions/platform-specific/index.native.tsx]: this should just be a chainAction of `osNetworkStatusChanged`

Comment (chrisnojima) [reply shared/actions/platform-specific/index.native.tsx]: we prefer side effects to be chained based on actions vs having to explicitly plumb through calls

File: go/libkb/appstate.go [hunk 3]
```diff
@@ -85,6 +85,57 @@ func (a *MobileAppState) StateAndMtime() (keybase1.MobileAppState, *time.Time) {
...
+		ch <- a.state
+	} else {
+		a.updateChs = append(a.updateChs, ch)
+	}
+	return ch
+}
+
+// Update updates the current network state, and notifies any waiting calls
+// from NextUpdate
+func (a *MobileNetState) Update(state keybase1.MobileNetworkState) {
```
Comment (mmaxim) [inline go/libkb/appstate.go]: This guy should just have a `State` function like the app state thing that inspires it.

**Pull request merged**